Add training examples (#19)
iterorganization/IMAS-ParaView: docs/source/training_jorek.rst

.. _`training_jorek`:

JOREK Case
----------
In this example, we will visualize a JOREK disruption case. A number of JOREK disruption cases are available on the `confluence page <https://confluence.iter.org/display/IMP/The+JOREK+disruption+cases>`_. We will visualize the electron temperature from the ``plasma_profiles`` IDS and the corresponding current magnitude in the inner vacuum vessel the ``wall`` IDS. We will create an animation to visualize how these change over time.

It is recommended to go through the steps below by hand, however if you do not want to follow the steps, a ParaView state file containing this example is made available :download:`here <pv_states/jorek_state.pvsm>`.

Loading the Electron Temperature
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
In this subsection, we load the JOREK grid and visualize the electron temperatures on this grid.

.. |ico1| image:: images/rotate_axis.png

> #. Navigate to *Sources > IMAS Tools* and select the JOREK Reader.

Reviewer: Please briefly explain why we need a different reader (see Slack notes from dry-run)

> #. Enter the following URI in the ``Enter URI`` field of the JOREK reader plugin:

Reviewer: Since this already uses the user, pulse, run, etc. you could use the "legacy parameters" option here as well

   .. code-block:: bash

      imas:hdf5?user=public;pulse=112111;run=2;database=ITER_DISRUPTIONS;version=4

> #. Select the ``plasma_profiles/1`` IDS in the IDS/Occurrence dropdown menu.

Reviewer: Link to confluence on the meaning of the different occurrences?

#. Repeat the previous two steps for the GGDReader containing the wall currents.
#. The clip must be applied to the Temporal Interpolator filter instead. Right-click the clip filter, select ``Copy Pipeline``, select the temporal interpolator, and select ``Paste Pipeline``. Ensure the wall currents are selected again in the coloring section. The old clip filter can now be removed.
#. Verify that both temporal interpolators are working by opening *View > Time Manager* and checking if the two temporal interpolators have the same number of time steps and that the time steps are of equal size.
#. Create an animation of the JOREK electron temperature and wall currents over time. Place the objects in the viewpoints in the desired orientation for the video. To create a video, go to *File > Save Animation*, provide a directory and a name for the video, and select ``OK``.
#. In the pop-up window, video settings such as image resolution and compression can be changed. In this example, only change the frame rate to 10. Press ``OK`` to start generating the animation. This may take a while.

The resulting animation is shown below:

> .. figure:: images/training/jorek.gif

Reviewer: Maybe add a warning for photosensitive people that this image contains flickering lights?

Reviewer: Or fix the AVI export by excluding the strange frames :)

Author: I created a new GIF containing only the time steps 0.25-0.5 and lowered the frame rate to 5, that should help with the rapid flickering.

.. _`training_jorek`:

JOREK Case
----------

> In this example, we will visualize a JOREK disruption case. A number of JOREK disruption cases are available on the `confluence page <https://confluence.iter.org/display/IMP/The+JOREK+disruption+cases>`_. We will visualize the electron temperature from the ``plasma_profiles`` IDS and the corresponding current magnitude in the inner vacuum vessel of the ``wall`` IDS. We will create an animation to visualize how these change over time.

Reviewer: ```suggestion
In this example, we will visualize a JOREK disruption case. A number of JOREK disruption cases are available on `this confluence page <https://confluence.iter.org/display/IMP/The+JOREK+disruption+cases>`_ (behind login wall). We will visualize the electron temperature from the ``plasma_profiles`` IDS and the corresponding current magnitude in the inner vacuum vessel of the ``wall`` IDS. We will create an animation to visualize how these change over time.
```

You can download the ParaView state file for this example :download:`here <pv_states/jorek_state.pvsm>`. However, we recommend that you manually follow the steps outlined below.

Loading the Electron Temperature
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
In this subsection, we load the JOREK grid and visualize the electron temperature on this grid.

.. |ico1| image:: images/rotate_axis.png

> #. The JOREK grid uses a combined finite-element and Fourier-series discretization. Bicubic finite elements describe fields in the poloidal plane, while a Fourier series handles variation in the toroidal direction. The standard GGD reader cannot process this grid structure natively, therefore, the JOREK reader was made available to load the JOREK datasets. To load it, navigate to *Sources > IMAS Tools* and select the JOREK Reader.

Reviewer: ```suggestion
#. The JOREK grid uses a combined finite-element and Fourier-series discretization. Bicubic finite elements describe fields in the poloidal plane, while a Fourier series handles variation in the toroidal direction. The standard GGD reader cannot process this grid structure, therefore, the JOREK reader was made available to load JOREK datasets. To load it, navigate to *Sources > IMAS Tools* and select the JOREK Reader.
```

#. Set the normal vector to ``0, -1, 0`` and select ``Apply`` to apply the filter.
#. To distinguish between the wall currents and the electron temperature grid, change the wall current color map. Edit the color map, select ``Select a color map from default presets``, and choose a different color map.

.. figure:: images/training/jorek_wall_currents.png

   JOREK GGD showing the electron temperature surrounded by total current in the inner vacuum vessel (``t=0.309984``). Data provided by J. Artola.

Temporal Interpolation and Animation
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
In this subsection, we create an animation of the loaded electron temperature and wall currents. We interpolate the time basis of both plugins to a linearly spaced basis.

#. To visualize the current time in the video, add a time value in the corner of the viewer using *Sources > Annotation > Annotate Time*. Press ``Apply`` to apply the time annotation source.
#. Select the JOREKReader and apply a ``Temporal Interpolator`` filter found under *Filters > Temporal > Temporal Interpolator*.
#. Set the ``Discrete Time Step Interval`` to 0.01, and select ``Apply`` to apply the temporal interpolation.

> #. Now we will do the same for the wall currents, select the created Clip filter a ``Temporal Interpolator`` filter found under *Filters > Temporal > Temporal Interpolator*. 

Reviewer: This sentence is not complete

```suggestion
#. Now we will do the same for the wall currents. Select the created Clip filter and apply another ``Temporal Interpolator`` filter found under *Filters > Temporal > Temporal Interpolator*. 
```
